src/animation/mod.rs

```diff
@@ -37,6 +37,7 @@ struct PlayingAnimation2D {
     repeat: RepeatAnimation,
     speed: f32,
     elapsed: f32,
+    frame: usize,
```

Comment:
Fells wrong to have this on here considering we have all the information already here. But unless we ask the user to supply `Res<Assets<AnimationClip2D>>` this is the best we can do.

Reply:
Yeah I know what you mean, if you have ideas for a better implementation feel very free to edit this PR. I needed this in my game which is why I just took the easiest way.